docs/doxygen/bda.md

```diff
@@ -0,0 +1,147 @@
+  similar than measurements with long baselines.
+- The frequency. Measurements for low frequencies are less similar than 
+  for high
+  frequencies. (Is this statement correct???)
+
+## Current situation
+
+- A [DPBuffer](@ref DP3::DPPP::DPBuffer) has a regular multidimensional
+  cube structure:
+  - Main layout: 
+  - The length of a time step is equal for all baselines.
+  - A DPBuffer contains the measurements for a single time step.
+  - The amount of frequency channels is equal for all baselines.
+    All channels have the same size.
+- The [DPStep::process()](@ref DP3::DPPP::DPStep::process) function has a
+  single [DPBuffer](@ref DP3::DPPP::DPBuffer) argument. It processes
+  the measurements for a single time step.
+
+## Requirements
+
+DP3 should support the following:
+1. Converting a regular buffer to BDA data, using a BDA algorithm.
+2. Converting BDA data to a regular buffer, using an expansion algorithm.
+3. Read /write BDA data from/to an MS using the Casacore library.
+   The MS format should be aligned with other tools.
+   (Which tools???)
+4. Perform processing steps that only use the data for the current iteration:
+   - Predict
+   - ApplyBeam
+   - ApplyCal
+   - Preflagger
+   - Phaseshift
+   - Filter
+   - Scaledata
+   - Preflagger
+   - UVWFlagger
+5. Perform processing steps that use data from multiple iterations.
+   The steps store data items of the current iteration for
+   use in later iterations. These steps include:
+   - Calibrate
+   - DDECal
+6. The DPPP input should allow specifying the BDA operations above.
+
+### Non-requirements
+
+The following processing steps explicitly do have to support BDA data,
+since it is impossible:
+- Average
+- AOFlagger
+- Interpolate
+- StationAdder
+
+## Design
+
+The current [DPStep](@ref DP3::DPPP::DPStep) class can be reused for supporting
+BDA operations, by extending the process() method with support for BDA data.
+This approach supports all requirements: A step can perform a BDA process()
+call sending BDA data to its next step and implement the BDA process()
+method for receiving BDA data from its previous step.
+
+With this approach, the DPPP input format can remain the same: The steps
+will receive and send data in the requested format.
+
+### DPStep structure
+
+DPSteps can support BDA data using a new process() function for BDA data.
+The default implementatation for this function will throw an exception.
+The existing process() function for regular data, which is now abstract,
+should also get a default implementation that throws an exception. In code:
+
+    class DPStep {
+    public:
+      virtual bool process(DPBuffer) { throw exception; } // Process regular data.
+      virtual bool process(BDABuffer) { throw exception; } // Process BDA data.
+    };
+
+With these two process functions, a DPStep can support
+regular and/or BDA data: Existing DPSteps, which only support regular
+data will remain working. Adding support for BDA data can be done
```

Comment:
"will remain working" -> with minimal changes

Reply:
I don't see how existing step classes require minimal changes: They already override the process() function for regular buffers, so that functionality will remain working. They don't have to override the process() function for BDA buffers, since the default implementation handles it.

Reply:
I expect at least some renaming, e.g. Buffer should be renamed to RegularBuffer, etc.